src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -1158,6 +1144,27 @@ impl<T: 'static> EventProcessor<T> {
                 }
             }
             _ => {
+                if event_type == self.xkbext.first_event_id {
+                    let xev = unsafe { &*(identity(xev) as *const _ as *const ffi::XkbAnyEvent) };
+                    match xev.xkb_type {
+                        ffi::XkbStateNotify => {
+                            let xev =
+                                unsafe { &*(xev as *const _ as *const ffi::XkbStateNotifyEvent) };
+                            if matches!(xev.event_type, 2 | 3) {
+                                // TODO: Is `as u32` fine here?
+                                self.kb_state.update_modifiers(
+                                    xev.base_mods,
+                                    xev.latched_mods,
+                                    xev.locked_mods,
+                                    u32::from_ne_bytes(xev.base_group.to_ne_bytes()),
+                                    u32::from_ne_bytes(xev.latched_group.to_ne_bytes()),
+                                    u32::from_ne_bytes(xev.locked_group.to_ne_bytes()),
+                                )
```

Comment:
> Is `as u32` fine here?

Yes, and it is the idiomatic way of doing this. As written in the [reference](https://doc.rust-lang.org/reference/expressions/operator-expr.html#semantics):

> Casting between two integers of the same size (e.g. i32 -> u32) is a no-op